src/main/java/com/synopsys/integration/jenkins/polaris/extensions/buildstep/PolarisBuildStepWorkflow.java

```diff
@@ -0,0 +1,105 @@
+/**
+ * synopsys-polaris
+ *
+ * Copyright (c) 2020 Synopsys, Inc.
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.synopsys.integration.jenkins.polaris.extensions.buildstep;
+
+import java.io.IOException;
+import java.util.Objects;
+
+import com.synopsys.integration.exception.IntegrationException;
+import com.synopsys.integration.jenkins.extensions.ChangeBuildStatusTo;
+import com.synopsys.integration.jenkins.extensions.JenkinsIntLogger;
+import com.synopsys.integration.stepworkflow.StepWorkflow;
+import com.synopsys.integration.stepworkflow.StepWorkflowResponse;
+
+import hudson.model.AbstractBuild;
+import hudson.model.Result;
+
+public class PolarisBuildStepWorkflow {
```

Comment:
There seems to be an implied pattern here-- would it serve us to establish a contract for these StepWorkflow building classes, or would that simply be architecture for the sake of architecture?

Reply:
I wondered the same thing. I got hung up on the different returns types of the perform() method. But maybe there's a good option I haven't thought of.